src/foam/java/PropertyInfo.js

```diff
@@ -97,7 +97,7 @@ foam.CLASS({
             type: 'int',
             visibility: 'public',
             args: [ { name: 'o1', type: 'Object' }, { name: 'o2', type: 'Object' } ],
-            body: 'return compareValues(get_(o1), get_(o2));'
+            body: 'return (o1 instanceof ' + this.sourceCls.name + ' ? compareValues(get_(o1), get_(o2)) : compareValues(cast(o1), cast(o2)));'
```

Comment:
In JS, we have both compare() and comparePropertyValues().
See: https://github.com/foam-framework/foam2/blob/master/src/foam/core/Property.js#L179

I don't mind also having a compareToPropertyValue or comparePropertyToObj which compares a property value to an object if needed.